File: CS/PS/Site/leetcode/easy/1512.py
from typing import List
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
      result = 0
      length = len(nums)

      for i in range(length - 1):
        for j in range(i + 1, length):
          if nums[i] == nums[j]:
            result = result + 1
      return result
    numIdenticalPairs("", [1,2,3,1,1,3])

File: CS/PS/Site/leetcode/easy/1512.js
/**
 * @param {number[]} nums
 * @return {number}
 2중반복문돌면서 (내원소 제외)
 나랑 같은거있으면 +1
 */
var numIdenticalPairs = function (nums) {
  let map = {};
  let count = 0;
  nums.forEach((num) => { 
    if (map[num]) {
      count += map[num];
      map[num]++;
    }
    else
      map[num] = 1;
  })
  
  console.log(count);
  
  return count;
};

numIdenticalPairs([1, 2, 3, 1, 1, 3]);
numIdenticalPairs([1,1,1,1]);
numIdenticalPairs([1, 2, 3]);

